### PR TITLE
Pv2 2143 calculate save and display submission metrics at monthend

### PR DESCRIPTION
**Pull request by tom-gough**

### CODE REVIEW

File: src/SFA.DAS.Payments.Monitoring.Metrics.Function.AcceptanceTests/SFA.DAS.Payments.Monitoring.Metrics.Function.AcceptanceTests.csproj [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    
+
+    
+
+    
+
+    
+  </ItemGroup>
+
+  <ItemGroup>
```
Comment (carlbm) [inline src/SFA.DAS.Payments.Monitoring.Metrics.Function.AcceptanceTests/SFA.DAS.Payments.Monitoring.Metrics.Function.AcceptanceTests.csproj:23]: You could probably get rid of these

**Review by carlbm — CHANGES_REQUESTED** [review]

File: src/SFA.DAS.Payments.Monitoring.Metrics.Data.UnitTests/MetricsPersistenceDataContextTests/GetSubmissionSummaryMetricsTests.cs [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+
+            fixture.Assert_MatchingSubmissionSummaries_AreReturned(result);
+        }
+
+        [Test]
+        public async Task AndNoMatchingSummariesExist_ThenNullReturned()
+        {
+            var result = await fixture.GetSubmissionsSummaryMetrics();
+
+            Assert.True(result is null);
```
[the remark this answers is not in the document]
Comment (ChughAkshay) [reply src/SFA.DAS.Payments.Monitoring.Metrics.Data.UnitTests/MetricsPersistenceDataContextTests/GetSubmissionSummaryMetricsTests.cs:40]: Updated this on PV2-2161 branch

File: src/SFA.DAS.Payments.Monitoring.Metrics.Function.AcceptanceTests/SFA.DAS.Payments.Monitoring.Metrics.Function.AcceptanceTests.csproj (hunk 1 above)
Comment (hardik-desai-kcl) [reply src/SFA.DAS.Payments.Monitoring.Metrics.Function.AcceptanceTests/SFA.DAS.Payments.Monitoring.Metrics.Function.AcceptanceTests.csproj:23]: we shouldn't mess around with project file

File: src/SFA.DAS.Payments.Monitoring.Metrics.Data/MetricsPersistenceDataContext.cs [hunk 3]
```diff
@@ -111,5 +106,32 @@ public async Task Save(SubmissionSummaryModel submissionSummary, CancellationTok
                 throw;
             }
         }
+
+        public async Task SaveSubmissionsSummaryMetrics(SubmissionsSummaryModel submissionsSummary, CancellationToken cancellationToken)
+        {
+            var transaction = await Database.BeginTransactionAsync(cancellationToken).ConfigureAwait(false);
```
[the remark this answers is not in the document]
Comment (hardik-desai-kcl) [reply src/SFA.DAS.Payments.Monitoring.Metrics.Data/MetricsPersistenceDataContext.cs:112]: there are 2 other methods on that class that does exactly the same so lets leave it as is

**Review by carlbm — DISMISSED** [review]

**Review by hardik-desai-kcl — APPROVED** [review]

**Pull request merged**